test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/TagHelperComponentTagHelperTest.cs

```diff
@@ -356,5 +388,19 @@ public Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
                 return Task.CompletedTask;
             }
         }
+
+        private static ViewContext CreateViewContext()
+        {
+            var viewContext = Mock.Of<ViewContext>();
+            var httpContext = new DefaultHttpContext()
+            {
+                RequestServices = new ServiceCollection()
+                .AddSingleton<ITagHelperComponentPropertyActivator>(new TagHelperComponentPropertyActivator())
+                .BuildServiceProvider()
+            };
+
+            viewContext.HttpContext = httpContext;
```

Comment:
Just so you know you can do `var viewContext = Mock.Of<ViewContext>(vc => vc.HttpContext == httpContext)`